### PR TITLE
Introducing `infrastructure` type in base

### PR DESCRIPTION
**Pull request by jenningsanderson**
# Description

Extends `base` theme with an _infrastructure_ type which includes a number of features from Daylight, primarily for rendering a basemap. One way of introducing _more_ features from Daylight into Overture.

Examples: Bridges, Communication / Power Lines, Aerialways, and more!

# Reference

- Based primarily on the Daylight Earth Table's [`infrastructure`](https://daylightmap.org/earth/infrastructure.html) and [`transit`](https://daylightmap.org/earth/transit.html) themes.
- Logic has been converted to Airflow in the data-platform-tf: https://github.com/OvertureMaps/tf-data-platform/pull/141
- The actual SQL query for the infrastructure conversion is here: https://github.com/OvertureMaps/schema/pull/133

# Testing

*Brief description of the testing done for this change showing why you are confident it works as expected and does not introduce regressions. Provide sample output data where appropriate.* 

TODO.

# Checklist

1. [X] Add relevant examples.
2. [ ] ~Add relevant counterexamples.~ Not doing counter-examples for base
3. [x] Update in-schema documentation using plain English written in complete sentences, if an update is required.
4. [x] Update Docusaurus documentation, if an update is required.
5. [ ] Review change with Overture technical writer to ensure any advanced documentation needs will be taken care of, unless the change is trivial and would not affect the documentation.

# Documentation Website
[Docs preview for this PR.](https://dfhx9f55j8eg5.cloudfront.net/pr/126)

### CODE REVIEW

File: schema/base/infrastructure.yaml [hunk 1]
```diff
@@ -0,0 +1,103 @@
...
+          - dam
+          - manhole
+          - pier
+          - power
+          - tower
+          - transit
+      class:
+        description: Further classification of the infrastructure type
+        type: string
+        enum:
```
Comment (vcschapp) [inline schema/base/infrastructure.yaml:42]: We should sort this list! Makes it a lot easier to grok it as a human...

Comment (jenningsanderson) [reply schema/base/infrastructure.yaml:42]: Agree! Sorry, you caught me in the middle of updating that...

**Review by BoZivk — APPROVED** [review]

**Review by jwass — DISMISSED** [review]

**Review by BoZivk — APPROVED** [review]

**Review by TristanDiet-TomTom — APPROVED** [review]

File: schema/base/infrastructure.yaml [hunk 2]
```diff
@@ -0,0 +1,101 @@
...
+          - cable
+          - cable_distribution
+          - cantilever
+          - catenary_mast
+          - communication_line
+          - communication_pole
+          - communication_tower
+          - connection
+          - cooling
+          - covered
```
[the remark this answers is not in the document]
Comment (jenningsanderson) [reply schema/base/infrastructure.yaml:61]: Yeah — in this particular case, it's referring to "covered bridge" where it's subtype "bridge" class = "covered"

[the remark this answers is not in the document]
Comment (TristanDiet-TomTom) [reply schema/base/infrastructure.yaml:61]: Subtype, and class are both required, 'bridge' is both a subtype and class.
So I understand it is to support following combinations
- 'bridge' > "covered"
- 'bridge' > "bridge"

Is the class 'bridge' also intended to combine with other subtypes e.g. 'dam' > "bridge" or
would such a bridge be captured as a separate entity and is subtype "bridge" only reserved for class 'bridge'?

Is my understanding correct, that If we would want to to capture, e.g. 'bus_stop' with 'covered' that would not be possible with the current setup?

[the remark this answers is not in the document]
Comment (jwass) [reply schema/base/infrastructure.yaml:61]: Transportation is proposing an `is_covered` flag for road segments that are covered. Seems like a good opportunity to unify these and also add it to infrastructure - which I think also solves @TristanDiet-TomTom's concern by making it a separate flag.

@ylinzhi @brad-richardson

[the remark this answers is not in the document]
Comment (jenningsanderson) [reply schema/base/infrastructure.yaml:61]: If `is_covered` is being proposed as a top-level attribute in transportation, then _infrastructure_ should definitely borrow it — I like that. 

As a standalone column for `type=infrastructure` inside of _base_, I disagree with it as a column and would instead say we should just pass the `covered=yes` tag in the source_tags feature.

This is just to reiterate that at the end of the day, `base` is a non-schematized theme that should hold as little information as possible and instead encourage the development of future themes and useful schemas for features that will ideally "graduate" out of `base`.

[the remark this answers is not in the document]
Comment (TristanDiet-TomTom) [reply schema/base/infrastructure.yaml:61]: I'm aligned with adding `covered=yes` as a source_tags.
Is there then still a need to have bridge both as a subtype and class? 

- 'bridge' > "bridge"
- 'dam' > "bridge"
- ...?

**Review by jonahadkins — APPROVED** [review]

**Review by BoZivk — APPROVED** [review]

**Review by vcschapp — APPROVED** [review]

File: schema/base/infrastructure.yaml (hunk 2 above)
[the remark this answers is not in the document]
Comment (vcschapp) [reply schema/base/infrastructure.yaml:61]: Discussed in Schema TF 2024-03-27, agreed to merge for now. Can discuss more later.

**Review by TristanDiet-TomTom — APPROVED** [review]

**Pull request merged**